scripts/delete-unused-users.py

```diff
@@ -15,22 +15,18 @@
 https://discourse.jupyter.org/t/is-there-a-way-to-bulk-delete-old-users/20866/3
 """
 import argparse
-from datetime import timedelta, datetime
+import json
 import logging
 import os
 import requests
 import sys
 
+from datetime import timedelta, datetime
 from dateutil.parser import parse
 
 logging.basicConfig(stream=sys.stdout, level=logging.WARNING)
 logger = logging.getLogger(__name__)
 
-token = os.environ["JUPYTERHUB_API_TOKEN"]
-headers = {
-    "Accept": "application/jupyterhub-pagination+json",
-    "Authorization": f"Bearer {token}",
-}
```

Comment:
It is interesting that the script was getting its authorization from the user's environment. This would work on the hub that the user was logged into, so I wonder why the script didn't also deduce the hub to operate on in the same way rather than having the user pass in the hub as an argument.

Being able to operate on multiple hubs is an improvement.

Reply:
my understanding is that the script was initially meant to be run in a remote/non-hub environment.

YMMV.